sound/soc/sof/sof-priv.h

```diff
@@ -423,7 +426,8 @@ struct sof_ipc_pm_ops {
  */
 struct sof_ipc_fw_loader_ops {
 	int (*validate)(struct snd_sof_dev *sdev);
-	size_t (*parse_ext_manifest)(struct snd_sof_dev *sdev);
+	size_t (*parse_ext_manifest)(struct snd_sof_dev *sdev, const struct firmware *fw,
+				     u32 lib_index, u32 flags);
```

Comment:
The `lib_index` and the `flags` are only for IPC4 internal use and should not be exposed to upper, generic layer.
`ipc_ops->fw_loader->parse_ext_manifest()` is _only_ called with a static set of parameters, never with lib_index != 0 nor without the SOF_FW_PARSE_MANIFEST_PRE_BOOT flag.